lib/kafka/async_producer.rb

```diff
@@ -218,6 +218,12 @@ def run
             raise "Unknown operation #{operation.inspect}"
           end
         end
+      rescue Kafka::Error => e
```

Comment:
Just realized that you might want to make this `rescue => e` because it's still possible for some non-kafka error to crash the host while `thread.abort_on_exception = true` is still used.

Reply:
That would be a bug that would warrant a hard crash.

Reply:
Maybe for your workload but for us it's unacceptable. Would you be open to making this behavior configurable?

Reply:
Have you ever seen a non-Kafka error cause a crash? Do all other libraries you use rescue and hide all runtime errors?

Reply:
I'm with @dasch  on this one. @film42  can't you wrap around your logic with rescue?

Reply:
If you have very specific requirements for an async producer, it's also possible to create your own. The one included in ruby-kafka simply wraps the sync producer, so it's actually not that difficult.

Reply:
@mensfeld I don’t have access to this code path because it’s in a separate thread that will crash the VM if something isn’t rescued. 

@dasch This is the first time I’ve seen a library that uses abort on exception to find bugs. I don’t think you need to handle any possible error ever here, but if you’re saying “any unhabdled error will crash the VM,” then I think that warrants more protection. But you’re right, it’s not hard to make my own. I was hoping we could find a way to make the default one more safe and generic. Sorry, I’m not trying to be annoying or argumentative. Thanks for working through this :)

So in summary I think not aborting on exception but publishing out all errors, and using a back-off timeout would cover both of our cases and expose any library bugs to the user. And I’m not asking you to write this, I’m just wondering if you think it’s a good pattern or not. I’m more than happy to make a cut and open a PR. I know you prefer the current implementation, but is there some kind of scenario where you’d be happy to remove abort on exception?

Reply:
I think not aborting on unexpected exceptions is reasonable, as long as the error is being logged. The worker thread would die though, so all writes would be dropped. 

Let me push an update to the PR.

Reply:
I have some ideas on preserving writes between worker threads but as for now this branch performs the same as master branch (where the vm would stop and writes would be lost) so 👍  .